File: Python3/hamming_distance.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        bin_x = bin(x)[2:]
        bin_y = bin(y)[2:]
        
        if len(bin_x) < len(bin_y): bin_x = bin_x.zfill(len(bin_y))
        else: bin_y = bin_y.zfill(len(bin_x))
            
        hamm_dist = 0
        
        for i in range(len(bin_x)):
            if bin_x[i] != bin_y[i]: hamm_dist += 1
                
        return hamm_dist

File: C/hamming_distance.c
int* intTobin(int n) {
    int* bin = calloc(32, sizeof(int));
    int index = 31;
    
    while(n > 0) {    
        bin[index] = (n % 2);                
        n = n / 2;
        index --;
    }
         
    return bin;
}

int hammingDistance(int x, int y){
    int* bin_x = intTobin(x);
    int* bin_y = intTobin(y);
                
    int hamm_dist = 0;
        
    for(int i = 0; i < 32; i++) {
        if(bin_x[i] != bin_y[i]) 
            hamm_dist ++;
    }
        
    return hamm_dist;
}
